server/src/main/java/io/unitycatalog/server/persist/utils/FileUtils.java

```diff
@@ -1,178 +1,62 @@
 package io.unitycatalog.server.persist.utils;
 
-import com.amazonaws.auth.AWSStaticCredentialsProvider;
-import com.amazonaws.auth.BasicSessionCredentials;
-import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3ClientBuilder;
-import com.amazonaws.services.s3.model.ListObjectsRequest;
-import com.amazonaws.services.s3.model.ObjectListing;
-import com.amazonaws.services.s3.model.ObjectMetadata;
-import com.amazonaws.services.s3.model.PutObjectRequest;
 import io.unitycatalog.server.exception.BaseException;
 import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.service.credential.CredentialOperations;
+import io.unitycatalog.server.service.iceberg.FileIOFactory;
 import io.unitycatalog.server.utils.Constants;
 import io.unitycatalog.server.utils.ServerProperties;
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.net.URI;
-import java.nio.file.FileVisitOption;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.Comparator;
-import java.util.stream.Stream;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.InputFile;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+
 public class FileUtils {
   private static final Logger LOGGER = LoggerFactory.getLogger(FileUtils.class);
   private static final ServerProperties properties = ServerProperties.getInstance();
+  private static final CredentialOperations credentialOps = new CredentialOperations();
+  private static final FileIOFactory fileIOFactory = new FileIOFactory(credentialOps);
 
   private FileUtils() {}
 
   private static String getStorageRoot() {
-    return properties.getProperty("storageRoot");
+    // Use local tmp directory as default storage root
```

Comment:
Why did we change this?

Reply:
When the server storage root wasn't configured, it would just create a directory called "null" and perform all operations in that dir. This will atleast have all operations done in the tmp dir.